src/ControlSystem/ControllerPND.cpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ControlSystem/ControllerPND.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+namespace Controllers {
+
+template <size_t deriv_order>
+DataVector PND<deriv_order>::operator()(
+    const DataVector& timescales,
+    const std::array<DataVector, deriv_order + 1>& q_and_derivs,
+    const double q_dt, const double deriv_dt) const noexcept {
+  // helper lambda for computing the binomial coefficients
+  auto binomial = [](size_t N, size_t k) {
+    return falling_factorial(N, k) / factorial(k);
```

Comment:
I'm guessing N and k will be <4 in most cases, so the numerator and denominator will likely never individually overflow